src/main/java/org/candlepin/subscriptions/files/ProductProfileRegistry.java

```diff
@@ -76,8 +77,15 @@ void addProductProfile(ProductProfile profile) {
         profileProducts.forEach(x -> engProductIdToProfileMap
             .put(Integer.parseInt(x.getEngProductId()), profile));
 
-        Set<String> duplicateIds = profileProducts.stream()
-            .flatMap(x -> x.getSwatchProductIds().stream())
+        Stream<String> fromProducts = profileProducts.stream()
+            .flatMap(x -> x.getSwatchProductIds().stream());
+        Stream<String> fromRoles = profile.getSyspurposeRoles().stream()
+            .flatMap(r -> r.getSwatchProductIds().stream());
+        Set<String> swatchProdIds = Stream.of(fromProducts, fromRoles)
+            .flatMap(t -> t)
```

Comment:
I think you can use `Function::identity` here.

Reply:
DONE